py/emitinlinethumb.c

```diff
@@ -108,7 +108,7 @@ STATIC mp_uint_t emit_inline_thumb_count_params(emit_inline_asm_t *emit, mp_uint
             return 0;
         }
         const char *p = qstr_str(MP_PARSE_NODE_LEAF_ARG(pn_params[i]));
-        if (!(strlen(p) == 2 && p[0] == 'r' && p[1] == '0' + i)) {
+        if (!(strlen(p) == 2 && p[0] == 'r' && (unsigned char)p[1] == '0' + i)) {
```

Comment:
It would probably make more sense to cast `('0' + i)` to `char` instead.

Reply:
Sure, but then you you're relying on `i` never going higher than 127 - '0' and cause undefined behaviour. This might might be ok in this cause though. I'm fine either way.

Reply:
In this particular case, the RHS expression `'0' + i` is `mp_uint_t` right? So casting the LHS to `mp_uint_t` might make more sense.

Reply:
@jimmo yes, that would definitely work too. this promotion from `unsigned char` to the wider unsigned value happens automatically though, and is well-defined. but you're right casting directly to `mp_uint_t` might be clearer.